x-pack/plugins/security_solution/public/common/containers/use_search_strategy/index.test.ts

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useSearchStrategy } from './index';
+import { renderHook } from '@testing-library/react-hooks';
+
+import { useObservable } from '@kbn/securitysolution-hook-utils';
+
+jest.mock('../../../transforms/containers/use_transforms', () => ({
+  useTransforms: jest.fn(() => ({
+    getTransformChangesIfTheyExist: null,
+  })),
+}));
+
+const mockAddToastError = jest.fn();
+
+jest.mock('../../hooks/use_app_toasts', () => ({
+  useAppToasts: jest.fn(() => ({
+    addError: mockAddToastError,
+  })),
+}));
+
+jest.mock('@kbn/securitysolution-hook-utils');
+
+const useObservableHookResult = {
+  start: jest.fn(),
+  error: null,
+  result: null,
+  loading: false,
+};
+
+const userSearchStrategyProps = {
+  factoryQueryType: 'testFactoryQueryType',
```

Comment:
I would recommend we do ```factoryQueryType: 'testFactoryQueryType' as FactoryQueryTypes``` here, line 56, and line 185
and remove `any` when calling `useSearchStrategy`

Reply:
Good one, thanks!